prow/pubsub/subscriber/subscriber.go

```diff
@@ -154,16 +172,70 @@ func (peh *periodicJobHandler) getProwJobSpec(cfg *config.Config, pe ProwJobEven
 }
 
 // presubmitJobHandler implements jobHandler
-type presubmitJobHandler struct{}
+type presubmitJobHandler struct {
+	GitHubClient githubClient
+	GitClient    git.ClientFactory
+}
+
+func (prh *presubmitJobHandler) getProwJobSpec(cfg prowCfgClient, pe ProwJobEvent) (*v1.ProwJobSpec, map[string]string, error) {
+	// presubmit jobs require Refs and Refs.Pulls to be set
+	refs := pe.Refs
+	if refs == nil {
+		return nil, nil, errors.New("Refs must be supplied")
+	}
+	if len(refs.Pulls) != 1 {
+		return nil, nil, fmt.Errorf("expect 1 pull, got: %d", len(refs.Pulls))
```

Comment:
I don't think there is any need to forbid batches.

Reply:
Agreed, allowed